tests/bdd/features/rebuild/test_log_based_rebuild.py

```diff
@@ -272,38 +249,39 @@ def log_based_rebuild_starts_after_some_time():
     childlist = target["children"]
     assert len(childlist) == NUM_VOLUME_REPLICAS
     for child in childlist:
-        if child["state"] == ChildState("Faulted") or child["state"] == ChildState(
-            "Degraded"
-        ):
-            assert child["uri"] == faulted_child_uri
-            assert child["rebuild_progress"] != 0
+        if child["uri"] == faulted_child_uri:
+            assert child["state"] == ChildState("Degraded")
+            assert child["state_reason"] == ChildStateReason("OutOfSync")
 
 
 @then("the node hosting rebuilding replica crashes permanently")
-def _():
+def the_node_hosting_rebuilding_replica_crashes_permanently():
     """the node hosting rebuilding replica crashes permanently."""
-    pass
-
-
-@then("the replica is not online again within the timeout period")
-def _():
-    """the replica is not online again within the timeout period."""
-    pass
+    Docker.kill_container(NODE_2_NAME)
+    check_child_removal()
 
 
-@then("the volume target moves to a different node before rebuild completion")
-def _():
-    """the volume target moves to a different node before rebuild completion."""
-    pass
+@retry(wait_fixed=500, stop_max_attempt_number=2)
+def check_child_removal():
+    nexus = ApiClient.nexuses_api().get_node_nexuses(NODE_1_NAME)
```

Comment:
Should use volume api, and get volume target

Reply:
In this case we are killing rebuilding container. We know nexus will be on this node and it should have only 1 nexus. So thought of using it.